include/openmc/source.h

```diff
@@ -73,27 +101,25 @@ class IndependentSource : public Source {
 
   // Properties
   ParticleType particle_type() const { return particle_; }
-  double strength() const override { return strength_; }
 
   // Make observing pointers available
   SpatialDistribution* space() const { return space_.get(); }
   UnitSphereDistribution* angle() const { return angle_.get(); }
   Distribution* energy() const { return energy_.get(); }
   Distribution* time() const { return time_.get(); }
 
-private:
-  // Domain types
-  enum class DomainType { UNIVERSE, MATERIAL, CELL };
+protected:
+  // Note that this method is not used since IndependentSource directly
+  // overrides the sample method
+  SourceSite sample_no_rejection(uint64_t* seed) const override { return {}; }
```

Comment:
Moreover, I find it funny that we've taken the base class route here, so ideally `IndependentSource` should NOT override `sample`. That way it retains the ability to do source site rejection.

It appears to me that this approach does not allow source site rejection with `IndependentSource`. That's no good! The `sample` method in the base class should NOT be virtual. Rather, base classes should only fill out the proposal distribution. This way we guarantee rejection functionality on any source class.